### PR TITLE
fix: plugin loading from layers

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description

Adds a pluginloader and layers stage to handle the override of the provider command line options based on the plugins defined in the CMDB layers

## Motivation and Context

The providers command line option wasn't being updated before the bootstrap process ran includeProviders. As a result of this the providers that were sources from layers weren't being included as part of the load process

## How Has This Been Tested?

Tested locally 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions

- [x] None

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

**Comment by RossMurr4y** [comment]
Just double checking that this addresses #1595?

**Review by RossMurr4y — DISMISSED** [review]

File: providers/shared/inputseeders/shared/id.ftl [hunk 1]
```diff
@@ -247,6 +253,23 @@
     ]
 [/#function]
 
+[#-- Define and setup the layer data so that it can be used by the plugin transformer --]
+[#function shared_configseeder_layers filter state]
```
Comment (RossMurr4y) [inline providers/shared/inputseeders/shared/id.ftl:257]: Are we missing a configseeder or does this just need a name change?

```
{
[2021-03-24T23:36:49.186Z]   "Timestamp": "2021-03-24T23:36:48.597Z",
[2021-03-24T23:36:49.186Z]   "Severity": "fatal",
[2021-03-24T23:36:49.186Z]   "Message": "Unable to invoke any of the input step function options for Config pipeline - check function naming",
[2021-03-24T23:36:49.186Z]   "Context": {
[2021-03-24T23:36:49.186Z]     "Id": "shared",
[2021-03-24T23:36:49.186Z]     "Priority": 100,
[2021-03-24T23:36:49.186Z]     "Sources": [
[2021-03-24T23:36:49.186Z]       "*"
[2021-03-24T23:36:49.186Z]     ],
[2021-03-24T23:36:49.186Z]     "Function": "configseeder"
[2021-03-24T23:36:49.186Z]   },
[2021-03-24T23:36:49.186Z]   "Detail": [
[2021-03-24T23:36:49.186Z]     [
[2021-03-24T23:36:49.186Z]       "shared",
[2021-03-24T23:36:49.187Z]       "configseeder",
[2021-03-24T23:36:49.187Z]       "fixture",
[2021-03-24T23:36:49.187Z]       "mock"
[2021-03-24T23:36:49.187Z]     ],
[2021-03-24T23:36:49.187Z]     [
[2021-03-24T23:36:49.187Z]       "shared",
[2021-03-24T23:36:49.187Z]       "configseeder",
[2021-03-24T23:36:49.187Z]       "fixture"
[2021-03-24T23:36:49.187Z]     ],
[2021-03-24T23:36:49.187Z]     [
[2021-03-24T23:36:49.187Z]       "shared",
[2021-03-24T23:36:49.187Z]       "configseeder"
[2021-03-24T23:36:49.187Z]     ]
[2021-03-24T23:36:49.187Z]   ]
[2021-03-24T23:36:49.187Z] }
```

Comment (roleyfoley) [reply providers/shared/inputseeders/shared/id.ftl:271]: I think that was from #1597 I've rebased now so it should be fine

**Comment by roleyfoley** [comment]
Still seems to be an outstanding issue, but I have NFI why 

when using the mock provider layers don't appear to be refreshed when the includeLayers step is run in the boostrap process. As a result of this setContext checks which look at the activeLayerState are failing.

**Comment by RossMurr4y** [comment]
With everything rebased after the last lot of merges, I get the following error when running the tests locally. I assume this is what you are getting too @roleyfoley 

```bash
fix-plugin-loading-seeder└─ $ ./test/run_shared_template_tests.sh 
###############################################
# Running template tests for the AWS provider #
###############################################
 -Output Dir: /home/hamlet/cmdb/hamlet/engine/core/hamlet_tests

--- Generating Management Contract ---

(Info) Generating outputs:
(Info)  - generationcontract
(Fatal) ! Exceptions occurred during template generation. Details follow...

{
  "Timestamp": "2021-03-26T00:25:03.541Z",
  "Severity": "fatal",
  "Message": "Could not find layer",
  "Context": {
    "type": "Product",
    "ActiveLayers": {}
  }
}
```

**Comment by roleyfoley** [comment]
Yep that's the one. The layer data is loaded in most places, but there seems to be somewhere that the layer data can't be found. I think @ml019 Is going to take a look

**Comment by roleyfoley** [comment]
Closing as this as been handled in #1605

**Pull request closed without merging**